nntrainer/tensor/half_tensor.h

```diff
@@ -35,6 +35,36 @@ class HalfTensor : public TensorBase {
    */
   HalfTensor(std::string name_ = "", Tformat fm = Tformat::NCHW);
 
+  /**
+   * @brief Construct a new HalfTensor object
+   *
+   * @param d
+   * @param alloc_now
+   * @param init
+   * @param name
+   */
+  HalfTensor(const TensorDim &d, bool alloc_now,
+             Initializer init = Initializer::NONE, std::string name = "");
+
+  /**
+   * @brief Construct a new HalfTensor object
+   *
+   * @param d
+   * @param buf
+   */
+  HalfTensor(const TensorDim &d, const void *buf = nullptr);
+
+#ifdef ENABLE_FP16
```

Comment:
I think this compiler preprocessor is not needed in code. This ```half_tensor.h``` can be only included if enable_fp16 is enabled and it will be controlled in meson level.

Reply:
Fixed.